### ISSUE TITLE
detekt-test 2.0.0-alpha.4 cannot resolve detekt-api test fixtures

### ISSUE THREAD

**Issue by sya-ri** [detekt/detekt#9409]
## Expected Behavior

`dev.detekt:detekt-test:2.0.0-alpha.4` should resolve from Maven Central in a normal Gradle project, the same way `2.0.0-alpha.3` does.

Projects that depend on `detekt-test` to test custom detekt rules should be able to upgrade from `2.0.0-alpha.3` to `2.0.0-alpha.4` without adding unpublished artifacts or custom metadata workarounds.

## Observed Behavior

Resolving the runtime classpath for `dev.detekt:detekt-test:2.0.0-alpha.4` fails because its Gradle module metadata requests the `dev.detekt:detekt-api-test-fixtures` capability from `dev.detekt:detekt-api:2.0.0-alpha.4`:

```text
Could not resolve dev.detekt:detekt-api:2.0.0-alpha.4.
Required by:
    project ':libs:detekt-rules' > dev.detekt:detekt-test:2.0.0-alpha.4
 > No matching variant of dev.detekt:detekt-api:2.0.0-alpha.4 with capability 'dev.detekt:detekt-api-test-fixtures' was found.
```

The `detekt-test` metadata contains this runtime dependency:

```json
{
  "group": "dev.detekt",
  "module": "detekt-api",
  "version": { "requires": "2.0.0-alpha.4" },
  "requestedCapabilities": [
    {
      "group": "dev.detekt",
      "name": "detekt-api-test-fixtures"
    }
  ]
}
```

But `detekt-api:2.0.0-alpha.4` does not publish a runtime/API test fixtures variant. Its module metadata only contains `testFixturesSourcesElements` for that capability, and Maven Central has the sources jar but not the compiled test fixtures jar:

```text
https://repo1.maven.org/maven2/dev/detekt/detekt-api/2.0.0-alpha.4/detekt-api-2.0.0-alpha.4-test-fixtures.jar
# 404

https://repo1.maven.org/maven2/dev/detekt/detekt-api/2.0.0-alpha.4/detekt-api-2.0.0-alpha.4-test-fixtures-sources.jar
# 200
```

This appears to come from the combination of:

```kotlin
// detekt-test/build.gradle.kts
implementation(testFixtures(projects.detektApi))
```

and:

```kotlin
// detekt-api/build.gradle.kts
val javaComponent = components["java"] as AdhocComponentWithVariants
listOf(configurations.testFixturesApiElements, configurations.testFixturesRuntimeElements).forEach { config ->
    config.configure {
        javaComponent.withVariantsFromConfiguration(this) {
            skip()
        }
    }
}
```

## Steps to Reproduce

Create a minimal Gradle project that resolves `detekt-test` at runtime:

```kotlin
plugins {
    kotlin("jvm") version "2.4.0"
}

repositories {
    mavenCentral()
}

dependencies {
    testImplementation("dev.detekt:detekt-test:2.0.0-alpha.4")
}
```

Then resolve the test runtime classpath, for example:

```shell
./gradlew testClasses
```

or in a custom detekt rule module using `detekt-test`:

```kotlin
dependencies {
    compileOnly("dev.detekt:detekt-api:2.0.0-alpha.4")
    testImplementation("dev.detekt:detekt-test:2.0.0-alpha.4")
    testImplementation("dev.detekt:detekt-test-utils:2.0.0-alpha.4")
}
```

The same setup resolves with `2.0.0-alpha.3`.

## Context

This blocks upgrading custom detekt rule test modules from `2.0.0-alpha.3` to `2.0.0-alpha.4`.

In our project the failing module tests custom rules with `dev.detekt.test.lint`, `dev.detekt.test.TestConfig`, and `dev.detekt.test.utils.*`. The source change was only a version bump from `2.0.0-alpha.3` to `2.0.0-alpha.4`; Gradle fails before tests run because the runtime classpath cannot be resolved.

It looks like either:

- `detekt-api` should publish the compiled test fixtures variant needed by `detekt-test`, or
- `detekt-test` should not expose/request `testFixtures(projects.detektApi)` in published runtime metadata.

## Your Environment

* Version of detekt used: `2.0.0-alpha.4`
* Version of Gradle used (if applicable): `9.5.1`
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version: GitHub Actions Ubuntu 24.04 and macOS 26
* Link to your project (if it's a public repository):

**Comment by 3flex** [comment]
Caused by https://github.com/detekt/detekt/pull/9212 (see https://github.com/detekt/detekt/commit/ff24b7627804ad0c31fc9958e2318979ec62b3e9#diff-82eae8dd57564853861267b5aa50f08437cf924ae1577174c96a8330aa99d17bR8), @BraisGabin should the referenced code be moved from test fixtures to detekt-api's main source set (or main source set in another module), or should we start publishing detekt-api test fixtures?

**Comment by BraisGabin** [comment]
I think that we shouldn't publish detekt-api fixtures. I see two options:

1. We fat-jar detekt-api fixtures inside `detekt-test`
2. We remove the dependency between `detekt-test` and `detekt-api` fixture.

The first seems a bit more complicated but maybe it would be better? No idea. I'm creating a PR for 2 because it's quicker.

**Comment by sschuberth** [comment]
As a work-around, something like this works for me:

```kotlin
configurations.named("testRuntimeClasspath") {
    resolutionStrategy.dependencySubstitution {
        val detektVersion = libs.versions.detektPlugin.get()

        substitute(module("dev.detekt:detekt-api"))
            .using(module("dev.detekt:detekt-api:$detektVersion"))
            .because("detekt-api test-fixtures variant is not resolvable with current Gradle/JDK 25 variant matching")
    }
}
```
